### DISCUSSION TITLE
Nsclient crashes constantly

### DISCUSSION THREAD

**Discussion by hebagit** [mickem/nscp#807] [Q&A]
We are experiencing issues with the nscp.exe Nsclient++ service to crash all the time.

This happens every few time when Nagios Core runs a check that makes nsclient launch a PowerShell script. On some servers it seems to happen maybe 10–20 % of the time, while other servers have no major issues. There is nothing special in the scripts.  I have tested with a simple 'Write-Output "Test"' script with no arguments, and it still crashes.

It can happen on new and old software, new and old hardware, and at least Windows 10 and Windows Server 2019. Pretty sure we have 2012 and 2016 where it crashes as well. I have yet to understand why some servers are more stable than others, when they use the same scripts, the same calling methods, etc. On one server it might crash 30 times a day, while on another it doesn't crash in a month.

Are anyone else seeing similar? Are anyone else using lots of PowerShell scripts in nsclient?

![image](https://github.com/mickem/nscp/assets/17749769/a6c6f01b-0932-4e23-b845-866ef5f9b747)

![image](https://github.com/mickem/nscp/assets/17749769/9e05e236-ea3a-4faf-a449-3dba15893afb)

![image](https://github.com/mickem/nscp/assets/17749769/b6482424-4c69-4543-8cdb-a75e57a7fa1b)

**Comment by mickem** [comment]
Presumably fixed as the issue was fixed
